openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py

```diff
@@ -224,6 +239,13 @@ def save_block(self, block):
         if not self.authored_data_store.has_changes(block):
             return  # No changes, so no action needed.
 
+        if block._runtime_requested_version != LatestVersion.DRAFT:  # pylint: disable=protected-access
```

Comment:
If we've loaded the block with a numeric version instead of `LatestVersion.DRAFT`, this check will fail, even if that version number matches the latest draft's version number.

Reply:
I know. But generally if you're loading a specific version, I assume you're not planning to make changes. If you were planning to make changes, you'd always want to load LatestVersion.DRAFT to avoid race conditions.